kevm-pyk/src/kevm_pyk/foundry.py

```diff
@@ -727,6 +730,50 @@ def foundry_simplify_node(
     return foundry.kevm.pretty_print(res_term, unalias=False, sort_collections=sort_collections)
 
 
+def foundry_merge_nodes(
+    foundry_root: Path,
+    test: str,
+    node_ids: Iterable[NodeIdLike],
+    bug_report: bool = False,
+) -> None:
+    def check_cells_equal(cell: str, nodes: Iterable[KCFG.Node]) -> bool:
+        nodes = list(nodes)
+        if len(nodes) < 2:
+            return True
+        cell_value = nodes[0].cterm.cell(cell)
+        for node in nodes[1:]:
+            if cell_value != node.cterm.cell(cell):
+                return False
+        return True
+
+    node_ids = [int(node) for node in node_ids]
+    br = BugReport(Path(f'{test}.bug_report')) if bug_report else None
+    foundry = Foundry(foundry_root, bug_report=br)
+    proofs_dir = foundry.out / 'apr_proofs'
+    contract_name, test_name = test.split('.')
+    proof_digest = foundry.proof_digest(contract_name, test_name)
+    proof = APRProof.read_proof(proof_digest, proofs_dir)
```

Comment:
Should we be using `Proof.read_proof` instead, so that it correctly determines if it's an `APRProof` or an `APRBMCProof`?

I think for the engagement, we're using `APRBMCProof` instead of `APRProof`, so we should probably make sure it works with taht.

Reply:
Done.